services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/service/impl/SubnetService.java

```diff
@@ -98,4 +105,38 @@ public void buildSubnetStates(NetworkConfiguration networkConfig, UnicastGoalSta
             multicastGoalState.getGoalStateBuilder().addSubnetStates(subnetStateBuilder.build());
         }
     }
+
+    public void buildSubnetState (UnicastGoalState unicastGoalState, String subnetId) throws Exception
+    {
+        InternalSubnetPorts subnetEntity = subnetPortsCache.getSubnetPorts(subnetId);
+        if (subnetEntity != null) {
```

Comment:
@DavidLiu506 Why you need to create `buildSubnetState()`, it's pretty much same as `buildSubnetStates()`, except `multicastGoalState` parameter. If you really don't need `multicastGoalState` for `router_state` (I am not quite sure here), you can still use the original `buildSubnetStates()`, but just send `null` for `multicastGoalState` parameter.